### PR TITLE
WIP: add cobbler-completion script for enhancement and review.

### PR DESCRIPTION
**Pull request by jehutyy**
The script is "just" working as expected. It provides simple completion
over usually used options like system, distro, profile, image etc.

### CODE REVIEW

**Comment by watologo1** [comment]
Tested: Works.
But some more change(s)/cleanup should be added together with this:
This replaces: config/bash/cobbler_bash
right?
So this file together with the bash directory should vanish.

Or maybe better add this file here:
config/bash/cobbler.completion
and remove:
config/bash/cobbler_bash

This should get removed anyway:
setup.py:
"config/bash/cobbler_bash"
and the new completion file (afaik know) should get installated into a distro specific completion dir with a dependency on bash-completion (or at least Recommends: bash_completion in cobbler.spec? and it should get listed in the files section, e.g. see below:)?
On SUSE this would be:
/usr/share/bash-completion/completions/cobbler     (new file should end up here)

**Comment by watologo1** [comment]
@jehutyy : Do you mind to re-submit, please.
I would say, just push it and we can add enhancements or missing pieces on top..., but the directory name you introduced:
cobbler-completion/cobbler.completion
is really poor imo. I am open to other suggestions, but imo it would fit best to the already existing:
config/bash/
directory.
@jehutyy : If you are short in time, I can come up with a new pull request. I would embedd this file with your author tag to the new file path and some addtional cleanups I already mentioned...

**Comment by jehutyy** [comment]
Yeah I'm sorry I just got back to work, I'm going to be short in time this week to make the changes you've requested. I can push the script into the right directory though. Is that okay with you ?

**Comment by watologo1** [comment]
> Yeah I'm sorry I just got back to work, I'm going to be short in time this week to make the changes you've requested. I can push the script into the right directory though. Is that okay with you ?

Yes, that would be perfect.
I can give it another test/try and add missing bits on an on-top patch.
Thanks a lot for the nice stuff,

      Thomas

**Comment by jehutyy** [comment]
I pushed a change, this is my first rodeo so please tell me if anything is wrong.

File: config/bash/cobbler.completion [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        [profile]="--name --distro --kickstart --kopts --ksmeta --virt-file-size --virt-ram --virt-type --virt-cpus --virt-path --virt-bridge --server --parent"
+        [system]="--name --profile --mac --ip-address --hostname --kopts --ksmeta --kickstart --netboot-enabled/N --server --gateway --dns-name --static-routes --power-address --power-type --power-user --power-pass --power-id"
+        [repo]="--mirror --name --rpmlist --creatrepo-flags --keep-updated --priority --arch --mirror-locally --breed"
+        [image]="--name --arch --breed --comment --ctime --mtime --file --depth --image-type --network-count --os-version --owners --parent --kickstart --virt-auto-boot --virt-bridge --virt-cpus --virt-file-size --virt-disk-driver --virt-path --virt-ram --virt-type --clobber --in-place"
+        [mgmtclass]="--name --comment --packages --files"
+        [package]="--uid --depth --comment --ctime --mtime --owners --name --action --installer --version --clobber --in-place"
+        [file]="--uid --depth --comment --ctime --mtime --owners --name --is-dir --action --group --clobber --in-place"
+    )
```
[the remark this answers is not in the document]
Comment (xenlo) [reply config/bash/cobbler.completion]: Like import, other commands has options that we could add here. For example `buildiso` could be added too:
```suggestion
        [import]="--arch --breed --os-version --path --name --available-as --kickstart --rsync-flags --help"
        [buildiso]="--iso --profiles --systems --tempdir --distro --standalone --source --exclude-dns --mkisofs-opts --help"
```

**Comment by jehutyy** [comment]
Thanks a lot for your review, I've taken your suggestions into account  and committed them.

**Comment by xenlo** [comment]
> 
> 
> Thanks a lot for your review, I've taken your suggestions into account and committed them.

Great!
Please note that you missed [one suggestion](https://github.com/cobbler/cobbler/pull/2208/files/#r351768313). It wasn't obvious as I reply with a second suggestion on the same line in place to update the previous one.
So ensure that you have the list of opts for `buildiso` as you added it in the `case` list.

Did you try those changes on an up to date version of cobbler? at least a version 3.something?

Merci,

File: config/bash/cobbler.completion (hunk 1 above)
Comment (xenlo) [inline config/bash/cobbler.completion]: Take care of the [second suggestion](https://github.com/cobbler/cobbler/pull/2208/files/7bd2d8308c4e4066c174721ee7c3432c8d3ebd04#r351768313).

**Comment by jehutyy** [comment]
I tested on a 2.8.5. The version provided with CentOS 7.7 currently. I'm about to test with the current master branch on github.

File: config/bash/cobbler.completion (hunk 1 above)
Comment (jehutyy) [reply config/bash/cobbler.completion]: I'm not sure, but it seems fine now, I mark this as resolved.

Comment (xenlo) [reply config/bash/cobbler.completion]: yes it's fine indeed

**Comment by watologo1** [comment]
> Did you try those changes on an up to date version of cobbler? at least a version 3.something?
Yes, I did.
I'll wait until this gets pushed and will try again and possibly try to identify and fix package building related issues for setup.py or cobbler.spec.
Thanks @jehutyy, thanks @xenlo , nice teamwork!

**Comment by watologo1** [comment]
I added a patch which adjusts setup.py (remove the old and add the new script).
This should calm down travis. I tried an external build, but I am not sure it's the last travis issue.
@jehutyy I cloned your repo, added my stuff on top and re-submitted here:
https://github.com/cobbler/cobbler/pull/2213

In case everything is green, I'll merge. Hope you are ok with this. Hm, if the patches are stashed, you might get lost as author? I try hard that this will not happen.

**Comment by watologo1** [comment]
Run through...
I resubmit with author of my patch changed to:
jehutyy <pagnat7@hotmail.com>

According to:
https://github.com/isaacs/github/issues/1368
I should be the committer jehutyy should be preserved as the author (of all patches, but this is intended). I'll give it  a try.

**Comment by watologo1** [comment]
resubmitted with a tiny on top fix:
https://github.com/cobbler/cobbler/pull/2214

**Pull request closed without merging**